### PR TITLE
Use LDBVersionedStore for wallet also

### PR DESCRIPTION
**Pull request by kushti**
This PR builds on top #969 . In addition the PR #969 , this one is using LDBVersionedStore for wallet registry as well. Old class VersionedLDBKVStore currently is used in benchmarks only , and to be removed.

Some ByteArrayWrapper usages eliminated as well, and IODB with ByteArrayWrapper are to be removed in next versions as well.

### CODE REVIEW

File: avldb/build.sbt [hunk 1]
```diff
@@ -16,8 +16,7 @@ libraryDependencies ++= Seq(
 )
 
 libraryDependencies ++= Seq(
-  ("org.fusesource.leveldbjni" % "leveldbjni-all" % "1.8").exclude("org.iq80.leveldb", "leveldb"),
-  "org.iq80.leveldb" % "leveldb" % "0.12"
+  "org.ethereum" % "leveldbjni-all"     % "1.18.3"
```
[the remark this answers is not in the document]
Comment (kushti) [reply avldb/build.sbt]: Fusesource's binaries are not maintained anymore. Ethereum's package is a fork of org.fusesource.leveldbjni upgraded to LevelDB 1.18  . Also binaries probably will work better for certain Windows versions (e.g. for 64-bits Windows Server 2019 it seems fusesource's native library is not working). Aside of Ethereum, it is used in Waves as well.

File: avldb/src/main/scala/scorex/db/KVStore.scala [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+  /**
+    * Read database element by its key
+    * @param key - key
+    * @return element if exists, None otherwise
+    */
+  def get(key: K): Option[V] = {
+    lock.readLock().lock()
+    val res = Option(db.get(key))
+    lock.readLock().unlock()
```
[the remark this answers is not in the document]
Comment (kushti) [reply avldb/src/main/scala/scorex/db/KVStore.scala]: fixed

File: avldb/src/main/scala/scorex/db/KVStore.scala [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    res
+  }
+
+
+  /**
+    * Iterate through the database to read elements according to a filter function
```
[the remark this answers is not in the document]
Comment (kushti) [reply avldb/src/main/scala/scorex/db/KVStore.scala]: fixed

File: avldb/src/main/scala/scorex/db/KVStore.scala [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+      }
+      bf.toIterator
+    } finally {
+      iter.close()
+      ro.snapshot().close()
+    }
+  }
+
+  /**
+    * Read all the database elements
```
[the remark this answers is not in the document]
Comment (kushti) [reply avldb/src/main/scala/scorex/db/KVStore.scala]: fixed

File: avldb/src/main/scala/scorex/db/LDBFactory.scala [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+
+import org.iq80.leveldb.{DB, Range, DBFactory, DBIterator, Options, ReadOptions, Snapshot, WriteBatch, WriteOptions}
+import scorex.util.ScorexLogging
+
+import scala.collection.mutable
+
+/**
+ * Registry of opened LevelDB instances.
+ * LevelDB prohibit access to the same storage file from more than one DB instance.
+ * And ergo application (mostly tests) quite frequently doesn't not explicitly close
```
[the remark this answers is not in the document]
Comment (kushti) [reply avldb/src/main/scala/scorex/db/LDBFactory.scala]: explicitly close is a perfect word order

File: avldb/src/main/scala/scorex/db/LDBFactory.scala [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+    } finally {
+      lock.writeLock().unlock()
+    }
+  }
+
+  def open(path: File, options: Options): DB = {
+    lock.writeLock().lock()
+    try {
+      add(path, factory.open(path, options))
+	} catch {
```
[the remark this answers is not in the document]
Comment (kushti) [reply avldb/src/main/scala/scorex/db/LDBFactory.scala]: fixed

File: avldb/src/test/scala/scorex/crypto/authds/avltree/batch/VersionedLDBAVLStorageSpecification.scala [hunk 7]
```diff
@@ -318,76 +315,32 @@ class VersionedIODBAVLStorageSpecification extends PropSpec
     blockchainWorkflowTest(prover)
   }
 
-  property("Persistence AVL batch prover (QuickStore backed) - blockchain workflow") {
-    quickTest {
-      val prover = createPersistentProverWithQuick()
-      blockchainWorkflowTest(prover)
-    }
-  }
-
```
[the remark this answers is not in the document]
Comment (kushti) [reply avldb/src/test/scala/scorex/crypto/authds/avltree/batch/VersionedLDBAVLStorageSpecification.scala]: QuickStore wasn't used outside tests, IODB will be removed soon as well

File: avldb/src/test/scala/scorex/crypto/authds/avltree/batch/benchmark/IODBBenchmark.scala [hunk 8]
```diff
@@ -11,31 +12,31 @@ object IODBBenchmark extends App with FileHelper {
...
+  val store = new LDBVersionedStore(getRandomTempDir, 10)
   val mods = generateModifications()
   var currentVersion: Option[Long] = None
 
   (0 until(NumMods, Step)) foreach { i =>
     println(i)
-    val mod: Seq[(ByteArrayWrapper, ByteArrayWrapper)] = mods.slice(i, i + Step)
-    val nextVersion = ByteArrayWrapper.fromLong(i)
+    val mod: Seq[(Array[Byte], Array[Byte])] = mods.slice(i, i + Step)
+    val nextVersion = Longs.toByteArray(i)
```
[the remark this answers is not in the document]
Comment (kushti) [reply avldb/src/test/scala/scorex/crypto/authds/avltree/batch/benchmark/IODBBenchmark.scala]: LDBVersionedStore has signatures similar to previously done VersionedLDBKVStore. So no new problems introduced hopefully, We can do wrappings in follow-up releases.

**Comment by kushti** [comment]
@aslesarenko thanks for the comments, I've addressed all of them!

File: avldb/src/main/scala/scorex/db/LDBFactory.scala (hunk 5 above)
[the remark this answers is not in the document]
Comment (aslesarenko) [reply avldb/src/main/scala/scorex/db/LDBFactory.scala]: I don't get the meaning of the sentence in this case.

**Pull request merged**